sls-versions/src/main/java/com/palantir/sls/versions/SlsVersionMatcher.java

```diff
@@ -164,14 +159,6 @@ public int compare(OrderableSlsVersion version) {
         return OrderableSlsVersion.safeValueOf(getValue());
     }
 
-    private static OptionalInt parseInt(String maybeInt) {
-        try {
-            return OptionalInt.of(Integer.parseInt(maybeInt));
-        } catch (NumberFormatException e) {
-            return OptionalInt.empty();
```

Comment:
I haven't read through the full PR, but I'm curious if you looked at Guava's [`Ints.tryParse(String)`](https://guava.dev/releases/19.0/api/docs/com/google/common/primitives/Ints.html#tryParse(java.lang.String)) which returns null instead of using expensive exceptions for control flow.

Reply:
Yep I think I would be happy to get that as a first-step, as the GC pressure is pretty savage at the moment. Only concern is that it would pull in a guava dep.  Here's a PR up anyway in case maintainers prefer it https://github.com/palantir/sls-version-java/pull/456

Reply:
I've approved the other PR, happy to merge that first if you like. Up to you if you'd like to further improve things with this PR -- perhaps worth benchmarking the difference?